crates/aranya-daemon/src/sync/task/quic/map.rs

```diff
@@ -0,0 +1,94 @@
+use std::{
+    collections::BTreeMap,
+    ops::{Deref, DerefMut},
+    sync::Arc,
+};
+
+use s2n_quic::{
+    application::Error as AppError,
+    connection::{Handle, StreamAcceptor},
+    Connection,
+};
+use tokio::sync::{self, mpsc, Mutex, RwLock};
+
+use crate::sync::task::quic::ConnectionKey;
+
+pub(crate) type ConnectionUpdate = (ConnectionKey, StreamAcceptor);
+type ConnectionMap = BTreeMap<ConnectionKey, Arc<Mutex<Handle>>>;
+
+pub(super) struct RwLockWriteGuard<'a, T: ?Sized> {
+    tx: mpsc::Sender<ConnectionUpdate>,
+    guard: sync::RwLockWriteGuard<'a, T>,
+}
+
+#[allow(clippy::expect_used, reason = "channel closed")]
+impl RwLockWriteGuard<'_, ConnectionMap> {
+    pub(super) async fn insert(
+        &mut self,
+        key: ConnectionKey,
+        conn: Connection,
+    ) -> (Option<Arc<Mutex<Handle>>>, Arc<Mutex<Handle>>) {
+        let (handle, acceptor) = conn.split();
+
+        let handle = Arc::new(Mutex::new(handle));
+        let existing = self.deref_mut().insert(key, Arc::clone(&handle));
+        self.tx.send((key, acceptor)).await.expect("channel closed");
+
+        (existing, handle)
+    }
+
+    pub(super) async fn remove(&mut self, key: ConnectionKey) {
+        if let Some(existing_conn) = self.deref_mut().remove(&key) {
+            // TODO: Use appropriate error code
+            existing_conn.lock().await.close(AppError::UNKNOWN);
+        }
+    }
+}
+
+impl<'a, T> Deref for RwLockWriteGuard<'a, T> {
+    type Target = sync::RwLockWriteGuard<'a, T>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.guard
+    }
+}
+
+impl<T> DerefMut for RwLockWriteGuard<'_, T> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.guard
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct SharedConnectionMap {
```

Comment:
The name of the struct should describe the purpose. The fact that it is implemented using a map is an implementation detail.
```suggestion
/// Stores active QUIC syncer connections with other peers.
pub(crate) struct QuicSyncerConnections {
```

Reply:
I think its probably fine, it does use a map, but the purpose is to be a specific map implementation.